pyresample/_spatial_mp.py

```diff
@@ -127,6 +131,14 @@ def __init__(self, projparams=None, preserve_units=True, **kwargs):
                     projparams = proj4_str_to_dict(projparams)
                 warnings.warn(warn_msg)
                 projparams = projparams.pop('init')
+            # New syntax 'EPSG:XXXX'
+            if 'EPSG' in kwargs or (isinstance(projparams, dict) and 'EPSG' in projparams):
```

Comment:
This is used to do Proj(EPSG=4326) and Proj({EPSG: 4326})? Is that a use case?

Reply:
@wroberts4 I modified the area def initialization so that `area.proj_dict={'EPSG': 'XXXX'}` if it is created via `AreaDefintion(projection='EPSG:XXXX')`. As for `EPSG=XXXX`: This happens for some reason when using `area_config.create_area_def()`.

Reply:
@sfinkens Does this mean that `create_area_def` is inconsistent with the regular creation of `AreaDefinition` (via `__init__`)?

Reply:
Not sure. `create_area_def` calls `Proj(proj_dict, ...)` whereas `AreaDefinition.__init__` calls `Proj(**proj_dict)`. According to the pyproj docs, both are valid.